tools/d2g/d2g.go

```diff
@@ -704,3 +705,19 @@ func (ril *RegistryImageLoader) ChainOfTrustCommand() string {
 		ril.Image.String(),
 	)
 }
+
+func hasGPUs() bool {
+	if testing.Testing() {
+		return true
+	}
+
+	out, err := host.Output("nvidia-smi", "--query-gpu=count", "--format=csv,noheader")
```

Comment:
If running this command not only checks GPUs are present, but actually causes them to be appear under `/dev` as a desired side-effect, we should definitely add a code comment here to explain that, or even change function name to something like `discoverGPUs` or `enableGPUs`. If running the command doesn't have side-effects like this, then maybe rename it to `hasNvidiaGPUs` since I think this is nvidia only...